src/ApiService/ApiService/Functions/Pool.cs

```diff
@@ -17,22 +18,27 @@ public class Pool {
     }
 
     [Function("Pool")]
-    public Async.Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymous, "GET", "POST", "DELETE", "PATCH")] HttpRequestData req)
-        => _auth.CallIfUser(req, r => r.Method switch {
-            "GET" => Get(r),
-            "POST" => Post(r),
-            "DELETE" => Delete(r),
-            "PATCH" => Patch(r),
-            var m => throw new InvalidOperationException("Unsupported HTTP method {m}"),
-        });
-
-    private async Task<HttpResponseData> Delete(HttpRequestData r) {
+    [Authorize(Allow.User)]
+    public Async.Task<HttpResponseData> Run(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "GET", "POST", "DELETE", "PATCH")]
+        HttpRequestData req,
+        FunctionContext context)
+        => req.Method switch {
+            "GET" => Get(req),
+            "POST" => Post(req, context),
+            "DELETE" => Delete(req, context),
+            "PATCH" => Patch(req, context),
+            _ => throw new InvalidOperationException("Unsupported HTTP method {m}"),
+        };
+
+    private async Task<HttpResponseData> Delete(HttpRequestData r, FunctionContext context) {
         var request = await RequestHandling.ParseRequest<PoolStop>(r);
         if (!request.IsOk) {
             return await _context.RequestHandling.NotOk(r, request.ErrorV, "PoolDelete");
         }
 
-        var answer = await _auth.CheckRequireAdmins(r);
+        var user = context.GetUserAuthInfo();
+        var answer = await _auth.CheckRequireAdmins(user);
         if (!answer.IsOk) {
```

Comment:
What do you think of:

1. Adding `CheckRequireAdmins` to the `Authorize` attribute
1. Splitting the endpoint into 2 definitions: `Get` and `Post, Delete, Patch`

Reply:
The benefit being that we'd be able to include that information in swagger docs.

Reply:
Yes, there is `[Authorize(Allow.Admin)]`. I didn't split any existing functions to reduce the changes here, but it's probably better to do this as part of the PR.

Reply:
Done this now.